custom_completions.go

```diff
@@ -249,11 +332,18 @@ func getFlagNameCompletions(flag *pflag.Flag, toComplete string) []string {
 		// Flag without the =
 		completions = append(completions, fmt.Sprintf("%s\t%s", flagName, flag.Usage))
 
-		if len(flag.NoOptDefVal) == 0 {
-			// Flag requires a value, so it can be suffixed with =
-			flagName += "="
-			completions = append(completions, fmt.Sprintf("%s\t%s", flagName, flag.Usage))
-		}
+		// Why suggest both long forms: --flag and --flag= ?
+		// This forces the user to *always* have to type either an = or a space after the flag name.
+		// Let's be nice and avoid making users have to do that.
+		// Since boolean flags and shortname flags don't show the = form, let's go that route and never show it.
+		// The = form will still work, we just won't suggest it.
+		// This also makes the list of suggested flags shorter as we avoid all the = forms.
+		//
```

Comment:
This deviates from bash completion.
I find it more user-friendly and don't understand why it was not done like that from the start.

Reply:
Yes oh yes. Showing two duplicate forms of the same flag adds visual clutter and makes it much harder to find what one needs. Thank you for this!

Reply:
I think this is the most user-friendly way, but I think it would be best to leave this as an option to the _Gen[Bash/Zsh/Fish]Completion_ function. With such an option each project could decide what it should be like for them.

Reply:
I don't really get the point of having an option to bring duplication (or the form with =) back. Completion should ease the use of the command. I would definitely hide the fact that there is the form with the equals character.

Reply:
> I think this is the most user-friendly way, but I think it would be best to leave this as an option to the Gen[Bash/Zsh/Fish]Completion function. With such an option each project could decide what it should be like for them.

But if we add an option, what would it be?  Should the project be able to enable both forms? Or maybe they just want the equal form instead of the non-equal form? So we'd need to support a three choice option.  I get the feeling this may be complicating things too much.

I suggest we start by removing the equal form and see if the community requests anything more.

Reply:
I haven't thought this out completely, but what if you use the `=` form only for flags that accept/require an option?
```
$ mycommand --<tab>
--binaryflag  --config-file=  --force --name= ...
```
Sometimes a program grows organically, and option names are not as intuitive as if they had all been designed on Day One. This could add immediate visual feedback to help a user figure out, say, "is it `--tmpdir` that takes a path option, or `--use-tmp`"? (contrived example)

Reply:
> Or maybe they just want the equal form instead of the non-equal form? So we'd need to support a three choice option. I get the feeling this may be complicating things too much.

Yes that would result in three choice option.
> I suggest we start by removing the equal form and see if the community requests anything more.

I'm fine with this. I was just pointing out that others might want it differently and having control  is not a bad idea.

Reply:
> what if you use the = form only for flags that accept/require an option?

I like that! The user would then immediately know when a flag value is required and when it is not.
My only hesitation is that I'm under the impression most people don't use the `=` form when adding a flag value; they may not like that we "force" them to use it, which comes back to @Luap99's point.

Let's wait until @jharshman weighs in before making any change (also, I don't want to touch the code anymore to facilitate a review).

Reply:
Personally I don't expect to see `=` signs in completions. It's an implicit understanding that you can use `=` signs in assignment but it is not required.